### PR TITLE
Power Armor Rework

### PR DESCRIPTION
**Pull request by BurpleBineapple**
## About The Pull Request

A bigger change to power armor's mechanics and projectile code. 

**TL;DR version: Projectiles with lower than 0.15 armor penetration now have a high chance to be completely blocked by power armor. Power armor's temperature values have also been fixed so that you don't die the second a cold/heat wave happens.**

Longer version of the whole thing, this PR has powered armors check for a projectile's armor penetration, and apply a block if it is lower than 0.15. Power armors have been given variables for a "block chance" and a "reflect chance".

How it works is that once you're hit by a projectile with the variable, the power armor runs a proc to check if the bullet is blocked or not. Salvaged armors have low values of about 20%. Functioning armors have values of 80% or higher. If the projectile is successfully blocked, it runs a parry and does one of two things. Either A: it deflects the projectile in a random direction based on the "deflect chance" or B: It stops the bullet and you take no damage.

The calibers this works on are:
- All 9mm except AP
- All 10mm except AP
- all 45ACP except AP
- all .22
- Regular shotgun buckshot
- Beanbag/Rubbershot shotgun rounds
- All hollow points (JHP)

357, 44, and 45-70 will work as normal. All rifles will work as normal. All lasers will work as normal. This new system means that power armor can no longer be reliably damaged by a lone wastelander with a 9mm Browning or a Grease Gun. AP bullets however will be **very** effective against power armor, encouraging their use if you plan to fight the Brotherhood of Steel.

Salvaged power armors also have these bonuses but to a much lesser extent, making the drastic speed debuff actually worth it now.

Later down the line I do plan to add a battery system for power armor as well, though that's a much harder task.

## Why It's Good For The Game

These values are taken from the Fallout 2 damage threshold and damage resistance values, which is how I selected which calibers should and shouldn't be blocked. The lowest power round that is guaranteed to go through power armor are 5.56 rifles and 357 handguns. Power armor is by no means invincible, but using low caliber bullets will not be very effective against the **pinnacle of pre-war military technology**. Rifles and AP rounds will still kill a Paladin in T-45d in less than a magazine, as normal.

Power armor will now truly be sought after, I would hope. Not just for the flex but for its superiority to normal armor. Even salvage will be worth the huge speed debuff.

## Changelog
:cl:
add: New variables to projectile code to set true or false to if power armor is able to block the bullet
add: New system for power armors to block low powered bullets based on an assigned percent chance
balance: Power armor is now stronger against pistol calibers, and encourages use of AP ammunition if fighting the Brotherhood of Steel
fix: Fixed power armor causing severe burns due to temperature changes
/:cl:

### CODE REVIEW

File: code/modules/clothing/suits/f13armor.dm [hunk 1]
```diff
@@ -329,14 +442,46 @@
...
+			return BLOCK_SUCCESS | BLOCK_PHYSICAL_INTERNAL
+	return ..()
+
 /obj/item/clothing/suit/armor/f13/power_armor/t51b
 	name = "T-51b power armor"
 	desc = "(X) The pinnacle of pre-war technology. This suit of power armor provides substantial protection to the wearer."
 	icon_state = "t51bpowerarmor"
 	item_state = "t51bpowerarmor"
 	slowdown = 0.15 //+0.1 from helmet = total 0.25
+	armor = list("tier" = 10, "energy" = 65, "bomb" = 62, "bio" = 100, "rad" = 99, "fire" = 90, "acid" = 0)
```
Comment (BurpleBineapple) [inline code/modules/clothing/suits/f13armor.dm:467]:
```suggestion
```

**Review by BurpleBineapple — COMMENTED** [review]
Somehow left the old armor value in, will delete

File: code/modules/projectiles/projectile.dm [hunk 2]
```diff
@@ -127,6 +127,10 @@
 	var/reflect_range_decrease = 5			//amount of original range that falls off when reflecting, so it doesn't go forever
 	var/is_reflectable = FALSE // Can it be reflected or not?
 
+	/*###DR ADDITIONS###*/
+	var/power_armor_blocks = FALSE //Does power armor stop this bullet?
```
Comment (MarinaGryphon) [inline code/modules/projectiles/projectile.dm:131]: This is just a bad idea. You should just use the `armour_penetration` var available on all subtypes of `/obj/item`. It ranges from 0 to 1; find a sufficient threshold to block bullets under. You could even set the threshold on each power armour subtype to give it varying amounts of armour toughness; some block anything under 80%, while some only block anything under 20%.

As it stands, adding an entirely new value to bullets is just silly.

File: code/modules/projectiles/projectile.dm [hunk 3]
```diff
@@ -836,6 +840,11 @@
 	var/obj/item/projectile/P = A
 	return istype(P) && P.is_reflectable
 
+//Bullet types that are blocked/deflected by power armor
+/proc/can_penetrate_power_armor(atom/A)
```
Comment (MarinaGryphon) [inline code/modules/projectiles/projectile.dm:844]: See above. This is, frankly, a silly proc as well; it's hyper-specific and all it does is run an istype and var access. If you still want to keep `power_armor_blocks` (you shouldn't), this should be either:
1) A proc on power armour, like `can_penetrate(atom/A)`, called in `run_block`. This is the ideal option.
or
2) Inlined in `run_block`, since it's just an istype and variable access:
```
	var/blocked = FALSE
	if(istype(A, /obj/item/projectile))
		var/obj/item/projectile/P = A
		blocked = P.power_armor_blocks
	if(blocked && ...)
```
etc.

File: code/modules/clothing/head/f13head.dm [hunk 4]
```diff
@@ -222,13 +222,20 @@
...
+	cold_protection = HEAD
+	min_cold_protection_temperature = FIRE_HELM_MIN_TEMP_PROTECT
 //	darkness_view = 128
 //	lighting_alpha = LIGHTING_PLANE_ALPHA_NV_TRAIT
 	var/on = FALSE
 	var/brightness_on = 5
 	var/power_on = 1
 	var/emped = 0
 	var/requires_training = TRUE
+	var/armor_block_chance = 0
```
Comment (MarinaGryphon) [inline code/modules/clothing/head/f13head.dm:236]: If you want to do the sane thing and change it to checking armour penetration on things it's hit with (as the var is available on anything derived from `/obj/item`), this should probably be replaced with
```suggestion
	var/armour_block_threshold = 0.75
```
or something similar.
Note that you should probably use a spreadsheet of AP values for this in order to fine-tune what guns hit what armours. It should feel much more natural if any extra ammo types are added, *and* it should also handle things that aren't subtypes of projectiles, i.e. a scrap launcher or something.

File: code/modules/clothing/head/f13head.dm [hunk 5]
```diff
@@ -222,13 +222,20 @@
...
+	var/list/protected_zones = list(BODY_ZONE_HEAD)
```
Comment (MarinaGryphon) [inline code/modules/clothing/head/f13head.dm:237]:
```suggestion
	var/const/list/protected_zones = list(BODY_ZONE_HEAD)
```
Not redefined on any subtypes, not permuted at runtime; this should be a const var.

File: code/modules/clothing/head/f13head.dm [hunk 6]
```diff
@@ -222,13 +222,20 @@
...
+	var/deflection_chance = 0
```
Comment (MarinaGryphon) [inline code/modules/clothing/head/f13head.dm:238]: I would probably rework this to be based on armour penetration; if something hits with high enough force and low enough armour penetration, that force should be redirected into a ricochet.

~~Alternatively, remove it entirely. Fallout 2 doesn't have deflection.~~

File: code/modules/clothing/suits/f13armor.dm [hunk 7]
```diff
@@ -202,8 +202,15 @@
...
 	strip_delay = 200
 	resistance_flags = LAVA_PROOF | FIRE_PROOF | ACID_PROOF
+	heat_protection = CHEST|GROIN|LEGS|FEET|ARMS|HANDS
+	max_heat_protection_temperature = FIRE_SUIT_MAX_TEMP_PROTECT
+	cold_protection = CHEST|GROIN|LEGS|FEET|ARMS|HANDS
+	min_cold_protection_temperature = FIRE_SUIT_MIN_TEMP_PROTECT
 	var/emped = 0
 	var/requires_training = TRUE
+	var/armor_block_chance = 0
+	var/list/protected_zones = list(BODY_ZONE_CHEST, BODY_ZONE_PRECISE_GROIN, BODY_ZONE_L_ARM, BODY_ZONE_R_ARM, BODY_ZONE_L_LEG, BODY_ZONE_R_LEG)
```
Comment (MarinaGryphon) [inline code/modules/clothing/suits/f13armor.dm:212]:
```suggestion
	var/const/list/protected_zones = list(BODY_ZONE_CHEST, BODY_ZONE_PRECISE_GROIN, BODY_ZONE_L_ARM, BODY_ZONE_R_ARM, BODY_ZONE_L_LEG, BODY_ZONE_R_LEG)
```
Not redefined on any subtypes, not permuted at runtime; this should be a const var.

File: code/modules/clothing/suits/f13armor.dm [hunk 8]
```diff
@@ -248,6 +255,22 @@
 	armor = list("tier" = 8, "energy" = 50, "bomb" = 48, "bio" = 60, "rad" = 50, "fire" = 80, "acid" = 0, "wound" = 40)
 	requires_training = FALSE
 	slowdown = 1.40
+	armor_block_chance = 25
+	deflection_chance = 10 //10% chance to block damage from blockable bullets and redirect the bullet at a random angle. Not nearly as effective as true power armor
+
+/obj/item/clothing/suit/armor/f13/power_armor/t45b/run_block(mob/living/owner, atom/object, damage, attack_text, attack_type, armour_penetration, mob/attacker, def_zone, final_block_chance, list/block_return)
```
Comment (MarinaGryphon) [inline code/modules/clothing/suits/f13armor.dm:261]: Duplicated code; use inheritance.

File: code/modules/clothing/suits/f13armor.dm [hunk 9]
```diff
@@ -257,6 +280,21 @@
 	armor = list("tier" = 8, "energy" = 50, "bomb" = 48, "bio" = 60, "rad" = 50, "fire" = 80, "acid" = 0, "wound" = 40)
 	requires_training = FALSE
 	slowdown = 1.40
+	armor_block_chance = 25
+	deflection_chance = 10 //10% chance to block damage from blockable bullets and redirect the bullet at a random angle. Not nearly as effective as true power armor
+
+/obj/item/clothing/suit/armor/f13/power_armor/ncr/run_block(mob/living/owner, atom/object, damage, attack_text, attack_type, armour_penetration, mob/attacker, def_zone, final_block_chance, list/block_return)
```
Comment (MarinaGryphon) [inline code/modules/clothing/suits/f13armor.dm:286]: Duplicated code; use inheritance.

File: code/modules/clothing/suits/f13armor.dm [hunk 10]
```diff
@@ -266,6 +304,21 @@
 	armor = list("tier" = 8, "energy" = 50, "bomb" = 48, "bio" = 60, "rad" = 50, "fire" = 80, "acid" = 0, "wound" = 40)
 	slowdown = 0.4
 	requires_training = FALSE
+	armor_block_chance = 20
+	deflection_chance = 5 //5% chance to block damage from blockable bullets and redirect the bullet at a random angle. Stripped down version of an already stripped down version
+
+/obj/item/clothing/suit/armor/f13/power_armor/raiderpa/run_block(mob/living/owner, atom/object, damage, attack_text, attack_type, armour_penetration, mob/attacker, def_zone, final_block_chance, list/block_return)
```
Comment (MarinaGryphon) [inline code/modules/clothing/suits/f13armor.dm:310]: Duplicated code; use inheritance.

File: code/modules/clothing/suits/f13armor.dm [hunk 11]
```diff
@@ -275,6 +328,21 @@
 	armor = list("tier" = 8, "energy" = 50, "bomb" = 48, "bio" = 60, "rad" = 50, "fire" = 80, "acid" = 0, "wound" = 40)
 	slowdown = 0.4
 	requires_training = FALSE
+	armor_block_chance = 20
+	deflection_chance = 5 //5% chance to block damage from blockable bullets and redirect the bullet at a random angle. Stripped down version of an already stripped down version
+
+/obj/item/clothing/suit/armor/f13/power_armor/hotrod/run_block(mob/living/owner, atom/object, damage, attack_text, attack_type, armour_penetration, mob/attacker, def_zone, final_block_chance, list/block_return)
```
Comment (MarinaGryphon) [inline code/modules/clothing/suits/f13armor.dm:334]: Duplicated code; use inheritance.

File: code/modules/clothing/suits/f13armor.dm [hunk 12]
```diff
@@ -283,6 +351,21 @@
 	item_state = "vaultpa"
 	armor = list("tier" = 8, "energy" = 50, "bomb" = 48, "bio" = 60, "rad" = 50, "fire" = 80, "acid" = 0)
 	slowdown = 0
+	armor_block_chance = 40
+	deflection_chance = 10 //10% chance to block damage from blockable bullets and redirect the bullet at a random angle. Not a heavy combat model
+
+/obj/item/clothing/suit/armor/f13/power_armor/vaulttec/run_block(mob/living/owner, atom/object, damage, attack_text, attack_type, armour_penetration, mob/attacker, def_zone, final_block_chance, list/block_return)
```
Comment (MarinaGryphon) [inline code/modules/clothing/suits/f13armor.dm:357]: Duplicated code; use inheritance.

File: code/modules/clothing/suits/f13armor.dm [hunk 13]
```diff
@@ -291,6 +374,21 @@
 	item_state = "excavator"
 	slowdown = 0.5 //+0.1 from helmet
 	armor = list("tier" = 8, "energy" = 60, "bomb" = 62, "bio" = 100, "rad" = 90, "fire" = 90, "acid" = 0)
+	armor_block_chance = 40
+	deflection_chance = 10 //10% chance to block damage from blockable bullets and redirect the bullet at a random angle. Not a heavy combat model
+
+/obj/item/clothing/suit/armor/f13/power_armor/excavator/run_block(mob/living/owner, atom/object, damage, attack_text, attack_type, armour_penetration, mob/attacker, def_zone, final_block_chance, list/block_return)
```
Comment (MarinaGryphon) [inline code/modules/clothing/suits/f13armor.dm:380]: Duplicated code; use inheritance.

File: code/modules/clothing/suits/f13armor.dm [hunk 14]
```diff
@@ -299,6 +397,21 @@
 	item_state = "t45dpowerarmor"
 	slowdown = 0.24
 	armor = list("tier" = 9, "energy" = 60, "bomb" = 62, "bio" = 100, "rad" = 90, "fire" = 90, "acid" = 0, "wound" = 60)
+	armor_block_chance = 80
+	deflection_chance = 20 //20% chance to block damage from blockable bullets and redirect the bullet at a random angle
+
+/obj/item/clothing/suit/armor/f13/power_armor/t45d/run_block(mob/living/owner, atom/object, damage, attack_text, attack_type, armour_penetration, mob/attacker, def_zone, final_block_chance, list/block_return)
```
Comment (MarinaGryphon) [inline code/modules/clothing/suits/f13armor.dm:403]: Duplicated code; use inheritance.

File: code/modules/clothing/suits/f13armor.dm [hunk 15]
```diff
@@ -329,14 +442,45 @@
 	item_state = "midwestgrey_pa"
 	armor = list("tier" = 9, "energy" = 60, "bomb" = 62, "bio" = 100, "rad" = 90, "fire" = 90, "acid" = 0, "wound" = 60)
 
+	armor_block_chance = 80
+	deflection_chance = 20 //20% chance to block damage from blockable bullets and redirect the bullet at a random angle
+
+/obj/item/clothing/suit/armor/f13/power_armor/midwest/run_block(mob/living/owner, atom/object, damage, attack_text, attack_type, armour_penetration, mob/attacker, def_zone, final_block_chance, list/block_return)
```
Comment (MarinaGryphon) [inline code/modules/clothing/suits/f13armor.dm:448]: Duplicated code; use inheritance.

File: code/modules/clothing/suits/f13armor.dm [hunk 16]
```diff
@@ -329,14 +442,45 @@
...
 	name = "T-51b power armor"
 	desc = "(X) The pinnacle of pre-war technology. This suit of power armor provides substantial protection to the wearer."
 	icon_state = "t51bpowerarmor"
 	item_state = "t51bpowerarmor"
 	slowdown = 0.15 //+0.1 from helmet = total 0.25
+	armor_block_chance = 85
+	deflection_chance = 35 //35% chance to block damage from blockable bullets and redirect the bullet at a random angle. Less overall armor compared to T-60, but higher deflection.
 	armor = list("tier" = 10, "energy" = 65, "bomb" = 62, "bio" = 100, "rad" = 99, "fire" = 90, "acid" = 0, "wound" = 70)
 
+/obj/item/clothing/suit/armor/f13/power_armor/t51b/run_block(mob/living/owner, atom/object, damage, attack_text, attack_type, armour_penetration, mob/attacker, def_zone, final_block_chance, list/block_return)
```
Comment (MarinaGryphon) [inline code/modules/clothing/suits/f13armor.dm:471]: Duplicated code; use inheritance.

File: code/modules/clothing/suits/f13armor.dm [hunk 17]
```diff
@@ -372,6 +516,22 @@
 	slowdown = 0.16
 	armor = list("tier" = 11, "energy" = 70, "bomb" = 82, "bio" = 100, "rad" = 100, "fire" = 95, "acid" = 0, "wound" = 80)
 
+	armor_block_chance = 90
+	deflection_chance = 20 //20% chance to block damage from blockable bullets and redirect the bullet at a random angle. Same deflection as T-45 due to it having the same general shape.
+
+/obj/item/clothing/suit/armor/f13/power_armor/t60/run_block(mob/living/owner, atom/object, damage, attack_text, attack_type, armour_penetration, mob/attacker, def_zone, final_block_chance, list/block_return)
```
Comment (MarinaGryphon) [inline code/modules/clothing/suits/f13armor.dm:522]: Duplicated code; use inheritance.

File: code/modules/clothing/suits/f13armor.dm [hunk 18]
```diff
@@ -396,6 +555,22 @@
 	item_state = "advpowerarmor1"
 	armor = list("linemelee" = 300, "linebullet" = 300, "linelaser" = 300, "energy" = 75, "bomb" = 72, "bio" = 100, "rad" = 100, "fire" = 90, "acid" = 0, "wound" = 90)
 
+	armor_block_chance = 100 //Enclave. 'nuff said
+	deflection_chance = 40 //40% chance to block damage from blockable bullets and redirect the bullet at a random angle. Your ride's over mutie, time to die.
+
+/obj/item/clothing/suit/armor/f13/power_armor/advanced/run_block(mob/living/owner, atom/object, damage, attack_text, attack_type, armour_penetration, mob/attacker, def_zone, final_block_chance, list/block_return)
```
Comment (MarinaGryphon) [inline code/modules/clothing/suits/f13armor.dm:561]: Duplicated code; use inheritance.

File: code/modules/clothing/head/f13head.dm [hunk 19]
```diff
@@ -292,33 +299,90 @@
...
 	desc = "(VIII) This power armor helmet is so decrepit and battle-worn that it have lost most of its capability to protect the wearer from harm. This helmet seems to be heavily modified, heavy metal banding fused to the helmet"
 	icon_state = "raiderpa_helm"
 	item_state = "raiderpa_helm"
 	armor = list("tier" = 8, "energy" = 50, "bomb" = 48, "bio" = 60, "rad" = 50, "fire" = 80, "acid" = 0, "wound" = 40)
-
 	requires_training = FALSE
+	armor_block_chance = 20
+	deflection_chance = 5
+
+/obj/item/clothing/head/helmet/f13/power_armor/raiderpa_helm/run_block(mob/living/owner, atom/object, damage, attack_text, attack_type, armour_penetration, mob/attacker, def_zone, final_block_chance, list/block_return)
```
Comment (MarinaGryphon) [inline code/modules/clothing/head/f13head.dm:329]: Duplicated code; use inheritance.

File: code/modules/clothing/head/f13head.dm [hunk 20]
```diff
@@ -292,33 +299,90 @@
...
 	desc = "(VIII) This power armor helmet is so decrepit and battle-worn that it have lost most of its capability to protect the wearer from harm."
 	icon_state = "t45hotrod_helm"
 	item_state = "t45hotrod_helm"
 	armor = list("tier" = 8, "energy" = 50, "bomb" = 48, "bio" = 60, "rad" = 50, "fire" = 80, "acid" = 0, "wound" = 40)
-
 	requires_training = FALSE
+	armor_block_chance = 20
+	deflection_chance = 5 //5% chance to block damage from blockable bullets and redirect the bullet at a random angle. Stripped down version of an already stripped down version
+
+/obj/item/clothing/head/helmet/f13/power_armor/hotrod/run_block(mob/living/owner, atom/object, damage, attack_text, attack_type, armour_penetration, mob/attacker, def_zone, final_block_chance, list/block_return)
```
Comment (MarinaGryphon) [inline code/modules/clothing/head/f13head.dm:352]: Duplicated code; use inheritance.

File: code/modules/clothing/head/f13head.dm [hunk 21]
```diff
@@ -292,33 +299,90 @@
...
 /obj/item/clothing/head/helmet/f13/power_armor/vaulttec
 	name = "Vault-Tec power helmet"
 	desc = "(VIII) A refined suit of power armour, purpose-built by the residents of Vault-115 in order to better keep the peace in their new settlement"
 	icon_state = "vaultpahelm"
 	item_state = "vaultpahelm"
 	armor = list("tier" = 8, "energy" = 50, "bomb" = 48, "bio" = 60, "rad" = 50, "fire" = 80, "acid" = 0, "wound" = 40)
+	armor_block_chance = 40
+	deflection_chance = 10 //10% chance to block damage from blockable bullets and redirect the bullet at a random angle. Not a heavy combat model
+
+/obj/item/clothing/head/helmet/f13/power_armor/vaulttec/run_block(mob/living/owner, atom/object, damage, attack_text, attack_type, armour_penetration, mob/attacker, def_zone, final_block_chance, list/block_return)
```
Comment (MarinaGryphon) [inline code/modules/clothing/head/f13head.dm:374]: Duplicated code; use inheritance.

**Comment by BurpleBineapple** [comment]
Need to fix a few things with messy code that someone pointed out. I actually may have had a stroke

File: code/modules/clothing/head/f13head.dm [hunk 22]
```diff
@@ -327,6 +391,21 @@
 	item_state = "t45dhelmet0"
 	actions_types = list(/datum/action/item_action/toggle_helmet_light)
 	armor = list("tier" = 9, "energy" = 60, "bomb" = 62, "bio" = 100, "rad" = 90, "fire" = 90, "acid" = 0, "wound" = 60)
+	armor_block_chance = 80
+	deflection_chance = 20 //20% chance to block damage from blockable bullets and redirect the bullet at a random angle
+
+/obj/item/clothing/head/helmet/f13/power_armor/t45d/run_block(mob/living/owner, atom/object, damage, attack_text, attack_type, armour_penetration, mob/attacker, def_zone, final_block_chance, list/block_return)
```
Comment (MarinaGryphon) [inline code/modules/clothing/head/f13head.dm:397]: Duplicated code; use inheritance.

File: code/modules/clothing/head/f13head.dm [hunk 23]
```diff
@@ -354,6 +433,21 @@
 	icon_state = "midwestgrey_helm"
 	item_state = "midwestgrey_helm"
 	armor = list("tier" = 9, "energy" = 60, "bomb" = 62, "bio" = 100, "rad" = 90, "fire" = 90, "acid" = 0, "wound" = 60)
+	armor_block_chance = 80
+	deflection_chance = 20 //20% chance to block damage from blockable bullets and redirect the bullet at a random angle
+
+/obj/item/clothing/head/helmet/f13/power_armor/midwest/run_block(mob/living/owner, atom/object, damage, attack_text, attack_type, armour_penetration, mob/attacker, def_zone, final_block_chance, list/block_return)
```
Comment (MarinaGryphon) [inline code/modules/clothing/head/f13head.dm:439]: Duplicated code; use inheritance.

File: code/modules/clothing/head/f13head.dm [hunk 24]
```diff
@@ -362,6 +456,21 @@
 	item_state = "t51bhelmet0"
 	armor = list("tier" = 10, "energy" = 65, "bomb" = 62, "bio" = 100, "rad" = 99, "fire" = 90, "acid" = 0, "wound" = 70)
 	actions_types = list(/datum/action/item_action/toggle_helmet_light)
+	armor_block_chance = 85
+	deflection_chance = 35 //35% chance to block damage from blockable bullets and redirect the bullet at a random angle. Less overall armor compared to T-60, but higher deflection.
+
+/obj/item/clothing/head/helmet/f13/power_armor/t51b/run_block(mob/living/owner, atom/object, damage, attack_text, attack_type, armour_penetration, mob/attacker, def_zone, final_block_chance, list/block_return)
```
Comment (MarinaGryphon) [inline code/modules/clothing/head/f13head.dm:462]: Duplicated code; use inheritance.

File: code/modules/clothing/head/f13head.dm [hunk 25]
```diff
@@ -396,6 +505,21 @@
 	item_state = "t60helmet0"
 	armor = list("tier" = 11, "energy" = 70, "bomb" = 82, "bio" = 100, "rad" = 100, "fire" = 95, "acid" = 0, "wound" = 80)
 	actions_types = list(/datum/action/item_action/toggle_helmet_light)
+	armor_block_chance = 90
+	deflection_chance = 20 //20% chance to block damage from blockable bullets and redirect the bullet at a random angle. Same deflection as T-45 due to it having the same general shape.
+
+/obj/item/clothing/head/helmet/f13/power_armor/t60/run_block(mob/living/owner, atom/object, damage, attack_text, attack_type, armour_penetration, mob/attacker, def_zone, final_block_chance, list/block_return)
```
Comment (MarinaGryphon) [inline code/modules/clothing/head/f13head.dm:511]: Duplicated code; use inheritance.

File: code/modules/clothing/head/f13head.dm [hunk 26]
```diff
@@ -407,13 +531,43 @@
 	icon_state = "excavator"
 	item_state = "excavator"
 	armor = list("tier" = 8, "energy" = 60, "bomb" = 62, "bio" = 100, "rad" = 90, "fire" = 90, "acid" = 0)
+	armor_block_chance = 40
+	deflection_chance = 10 //10% chance to block damage from blockable bullets and redirect the bullet at a random angle. Not a heavy combat model
+
+/obj/item/clothing/head/helmet/f13/power_armor/excavator/run_block(mob/living/owner, atom/object, damage, attack_text, attack_type, armour_penetration, mob/attacker, def_zone, final_block_chance, list/block_return)
```
Comment (MarinaGryphon) [inline code/modules/clothing/head/f13head.dm:537]: Duplicated code; use inheritance.

File: code/modules/clothing/head/f13head.dm [hunk 27]
```diff
@@ -407,13 +531,43 @@
...
 /obj/item/clothing/head/helmet/f13/power_armor/advanced
 	name = "advanced power helmet"
 	desc = "It's an advanced power armor Mk I helmet, typically used by the Enclave. It looks somewhat threatening."
 	icon_state = "advhelmet1"
 	item_state = "advhelmet1"
 	armor = list("linemelee" = 300, "linebullet" = 300, "linelaser" = 300, "energy" = 75, "bomb" = 72, "bio" = 100, "rad" = 100, "fire" = 90, "acid" = 0, "wound" = 90)
+	armor_block_chance = 100 //Enclave. 'nuff said
+	deflection_chance = 40 //40% chance to block damage from blockable bullets and redirect the bullet at a random angle. Your ride's over mutie, time to die.
+
+/obj/item/clothing/head/helmet/f13/power_armor/advanced/run_block(mob/living/owner, atom/object, damage, attack_text, attack_type, armour_penetration, mob/attacker, def_zone, final_block_chance, list/block_return)
```
Comment (MarinaGryphon) [inline code/modules/clothing/head/f13head.dm:559]: Duplicated code; use inheritance.

**Review by MarinaGryphon — CHANGES_REQUESTED** [review]

**Comment by BurpleBineapple** [comment]
Should be ready for review once more

**Comment by BurpleBineapple** [comment]
stiiiill some minor bugs with this one, so don't merge

**Comment by BurpleBineapple** [comment]
After further testing, it should be fine to merge. I was checking on an outdated branch.
